crates/db-store-rocksdb/src/checkpoint/db.rs

```diff
@@ -101,6 +101,48 @@ impl CheckpointDatabase for RBCheckpointDB {
     fn get_last_checkpoint_idx(&self) -> DbResult<Option<u64>> {
         Ok(rockbound::utils::get_last::<CheckpointSchema>(&*self.db)?.map(|(x, _)| x))
     }
+
+    fn get_latest_unproven_checkpoint_idx(&self) -> DbResult<Option<u64>> {
+        use strata_db::types::CheckpointProvingStatus;
+
+        // Use read transaction for consistency
+        self.db
+            .with_optimistic_txn(
+                rockbound::TransactionRetry::Count(self.ops.retry_count),
+                |txn| {
+                    let mut iterator = txn.iter::<CheckpointSchema>()?;
+                    iterator.seek_to_last();
+
+                    let mut latest_proven_idx = None;
+                    const MAX_SCAN: usize = 1000;
+
+                    for (scanned, item) in iterator.rev().enumerate() {
+                        if scanned >= MAX_SCAN {
+                            break;
+                        }
+
+                        let (idx, entry) = item?.into_tuple();
+                        if entry.proving_status == CheckpointProvingStatus::ProofReady {
+                            latest_proven_idx = Some(idx);
+                            break;
+                        }
+                    }
+
+                    // Use saturating_add to prevent overflow
+                    let next_to_prove = latest_proven_idx.map_or(0, |idx| idx.saturating_add(1));
+
+                    Ok::<_, anyhow::Error>(match txn.get::<CheckpointSchema>(&next_to_prove)? {
+                        Some(entry)
+                            if entry.proving_status == CheckpointProvingStatus::PendingProof =>
```

Comment:
why do we need this extra check?

shouldn't we have another status for _currently being proven_ proofs?